core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -848,15 +855,83 @@ public Graph build() {
+                    null != properties ? properties.getCacheServiceNamedViewSuffix(config.getGraphId()) : null);
+        }
+
+        private void updateNamedOperationResolverHook(final GraphConfig config) {
+            validateGetFromCacheHook(
+                    config,
+                    AddNamedOperation.class,
+                    NamedOperationResolver.class,
+                    null != properties ? properties.getCacheServiceNamedOperationSuffix(config.getGraphId()) : null);
+        }
+
+        /**
+         * This method is designed to keep suffix used for cache names aligned between
+         * a Handler which Adds to the cache and ResolverHook which Gets from the cache.
+         * If the suffix is mismatched then writing and reading to cache will not behave correctly.
+         *
+         * @param config               the graphConfig containing the Hooks
+         * @param operationClass       the Operation requiring cache write
+         * @param hookClass            the Hook requiring cache reading
+         * @param suffixFromProperties the suffix from property
+         * @see NamedOperationResolver#NamedOperationResolver(String)
+         * @see NamedViewResolver#NamedViewResolver(String)
+         */
+        private void validateGetFromCacheHook(final GraphConfig config, final Class<? extends Operation> operationClass, final Class<? extends GetFromCacheHook> hookClass, final String suffixFromProperties) {
+            if (store.isSupported(operationClass)) {
+                //Get Handler
+                final OperationHandler<Operation> addToCacheHandler = store.getOperationHandler(operationClass);
+                //If Handler is a AddToCacheHandler
+                final String suffix;
+                if (AddToCacheHandler.class.isAssignableFrom(addToCacheHandler.getClass())) {
+                    // get Suffix
+                    suffix = ((AddToCacheHandler<?>) addToCacheHandler).getSuffixCacheName();
+                } else {
+                    // otherwise get from properties
+                    LOGGER.warn(String.format(HANDLER_WAS_NOT_EXPECTED_TYPE_ADD_TO_CACHE_HANDLER, operationClass, AddToCacheHandler.class.getSimpleName(), suffixFromProperties));
+                    suffix = suffixFromProperties;
+                }
+                final List<GraphHook> hooks = config.getHooks();
+                //Is GetFromCacheHook missing
+                if (!hasHook(hooks, hookClass)) {
+                    //provide info about the graph not having the required hook
+                    LOGGER.info(HANDLER_WAS_SUPPLIED_BUT_WITHOUT_A_ADDING_WITH_SUFFIX, config.getGraphId(), operationClass, hookClass.getSimpleName(), hookClass.getSimpleName(), suffix);
+                    try {
+                        hooks.add(0, hookClass.getDeclaredConstructor(String.class).newInstance(suffix));
+                    } catch (final Exception e) {
+                        throw new RuntimeException(e);
```

Comment:
Does this not just mask the exception type?

Reply:
No, this will convert all checked exception into [unchecked exceptions](https://docs.oracle.com/javase/tutorial/essential/exceptions/runtime.html).

Reply:
So would it be better instead to add a `throws` at the start if the method?

Reply:
It would need to be caught somewhere else. I imagine that's not practical which is why it has been done like this.